### PR TITLE
NCL5075 Changes for dynamic groovy.

### PR DESCRIPTION
**Pull request by rnc**
This is draft for early feedback as its currently relying on a SNAPSHOT version of PME. 

This also has potential issues for repour parsing which needs thinking about.

~~( PR failing due to depending on a snapshot )~~

### CODE REVIEW

**Comment by geoand** [comment]
@rnc sorry I'm a little pressed for time, not sure I'll be able to get to this one today :(

**Comment by rnc** [comment]
@thescouser89  If I can get Travis working we can disable the other PR CI process.

**Comment by geoand** [comment]
Travis CI?

Cool!

File: build.gradle.kts [hunk 1]
```diff
@@ -87,11 +107,17 @@ subprojects {
             // Using non-deprecated archiveClassifier.set doesn't seem to work.
             @Suppress("DEPRECATION")
             classifier = ""
-            dependencies {
-                exclude(dependency("org.slf4j:slf4j-api:1.7.25"))
-            }
             // no need to add analyzer.init.gradle in the jar since it will never be used from inside the plugin itself
             exclude("analyzer-init.gradle")
+            // Minimise the resulting uber-jars to ensure we don't have massive jars
```
Comment (geoand) [inline build.gradle.kts]: :+1: :+1: :+1:

**Comment by geoand** [comment]
Looks really good!

I am assuming there will be a follow up with some docs on how the groovy stuff is used?

**Comment by rnc** [comment]
You want docs?! :-) 

Yeah, something akin to https://release-engineering.github.io/pom-manipulation-ext/guide/groovy.html (which also needs updating to note the deprecated `PMEInvocationPoint` annotation has been replaced by `InvocationPoint`.

**Comment by vibe13** [comment]
Looks good to me! Great! +1

**Pull request merged**